### ISSUE TITLE
add_histogram method calls report_surface with hard-coded iteration=0

### ISSUE THREAD

**Issue by vfdev-5** [allegroai/clearml#666]
Hi, 

we are wondering about this hard-coded value and especially that it is visible in UI. Specifying `step` value however does not display it anywhere...

https://github.com/allegroai/clearml/blob/90d060dd7e5f41e2359a6286902c05b5620d31a8/clearml/binding/frameworks/tensorflow_bind.py#L220

Context: https://github.com/pytorch/ignite/pull/2550#issuecomment-1115543004

Anything we can do with that or use some other more appropriate classes ? Thanks

**Comment by jkhenning** [comment]
Hi @vfdev-5,

This is indeed strange, I'll take a look

**Comment by eugen-ajechiloae-clearml** [comment]
Hi @vfdev-5! Iteration is hardcoded to 0 here because it is not relevant. All the histograms corresponding to a certain name are aggregated in one single surface that has the iteration of each corresponding histogram on the `yaxis`. In the surface that I saw in your related issue, the step would be here:
![image](https://user-images.githubusercontent.com/97950284/168488046-a3d10fd5-914f-422b-96c0-246df98796a5.png)
If we were to set the step there, there would be a lot of surfaces being displayed, each one with a higher iteration number including the others:
![image](https://user-images.githubusercontent.com/97950284/168488483-93480795-e3d6-4883-9d6e-9ad0a876cda4.png)

**Comment by vfdev-5** [comment]
@eugen-ajechiloae-clearml thanks for the answer! It makes sense. However, it would be nice to avoid duplicated info like "Iteration 0" in the title and the axis.

**Comment by eugen-ajechiloae-clearml** [comment]
@vfdev-5 Great news, we actually don't display that redundant `Iteration 0` anymore, this got fixed in the meantime
